src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -149,27 +150,30 @@ class DungeonFinderFeatures {
     @SubscribeEvent
     fun onItemTooltip(event: LorenzToolTipEvent) {
         if (!LorenzUtils.inSkyBlock) return
-        if (!config.coloredClassLevel) return
+
+        val coloredClass = config.coloredClassLevel
 
         val chestName = InventoryUtils.openInventoryName()
         if (chestName != "Party Finder") return
 
         val stack = event.itemStack
 
         val classNames = mutableListOf("Healer", "Mage", "Berserk", "Archer", "Tank")
-
         for ((index, line) in stack.getLore().withIndex()) {
             classLevelPattern.matchMatcher(line) {
                 val playerName = group("playerName")
                 val className = group("className")
                 val level = group("level").toInt()
                 val color = getColor(level)
-                event.toolTip[index + 1] = " §b$playerName§f: §e$className $color$level"
-                if(classNames.contains(className)) classNames.remove(className)
+                if(coloredClass) event.toolTip[index + 1] = " §b$playerName§f: §e$className $color$level"
+                classNames.remove(className)
             }
         }
+        if(!config.showMissingClasses) return
+        if(!stack.getLore()[0].removeColor().startsWith("Dungeon:")) return
```

Comment:
Added to prevent the classes from being shown on items such as "Create a party" or "Close"